src/generators/jsx-ast/utils/buildBarProps.mjs

```diff
@@ -0,0 +1,53 @@
+import readingTime from 'reading-time';
+import { visit } from 'unist-util-visit';
+import { DOC_API_BLOB_EDIT_BASE_URL } from '../../../constants.mjs';
+
+/**
+ * Builds sidebar navigation for API documentation pages
+ *
+ * @param {Map<string, Array<ApiDocMetadataEntry>>} groupedModules - Modules grouped by API
+ * @param {Array<ApiDocMetadataEntry>} headNodes - Main entry nodes for each API
+ */
+export const buildSideBarDocPages = (groupedModules, headNodes) =>
+  headNodes.map(node => {
+    const moduleEntries = groupedModules.get(node.api);
+
+    return {
+      title: node.heading.data.name,
+      doc: `${node.api}.html`,
+      headings: moduleEntries
+        .filter(entry => entry.heading?.data?.name && entry.heading.depth === 2)
+        .map(entry => [entry.heading.data.name, `#${entry.heading.data.slug}`]),
+    };
+  });
+
+/**
+ * Builds metadata for the sidebar and meta bar
+ *
+ * @param {ApiDocMetadataEntry} head - Main API metadata entry
+ * @param {Array<ApiDocMetadataEntry>} entries - All API metadata entries
+ */
+export const buildMetaBarProps = (head, entries) => {
+  // Extract text content for reading time calculation
+  const textContent = entries.reduce((acc, entry) => {
```

Comment:
Oh, I didn't know the resulting reduce would look like this, this can lead to asynchronous issues. You can use a method visitAll to get all these nodes in one array if all you want is to get a value. cc @avivkeller urgent.

Reply:
If it's urgent, can you open a fix?

Reply:
Can wait

Reply:
You are calling a callback inside another callback, these two are not synchronous. entries.reduce will not wait for visit() if visit() is async. Might not be the case here, but still, rather do a visitAll() and then grab all the values.

Reply:
`visit` isn't async (and `visitAll` isn't a function)